src/home/homectl.c

```diff
@@ -1763,6 +1763,32 @@ static int passwd_home(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int parse_disk_size(const char *t, uint64_t *ret) {
+        int r;
+
+        assert(t);
+        assert(ret);
+
+        if (streq(t, "min"))
+                *ret = 0;
+        else if (streq(t, "max"))
+                *ret = UINT64_MAX-1;  /* Largest size that isn't UINT64_MAX special marker */
+        else {
+                uint64_t ds;
+
+                r = parse_size(t, 1024, &ds);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parse disk size parameter: %s", t);
+
+                if (ds >= UINT64_MAX) /* UINT64_MAX has special meaning for us ("dont change"), refuse */
```

Comment:
UINT64_MAX-1 is special too, no? shouldn't that be refused too?

Reply:
No it's not special really. It's the largest of the non-special values, and that makes it a bit special if you so will ;-)

So if you grow to the home dir to the max you can just specify the largest valid size and we'll try our best to get there.